gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/publisher/CopyDataPublisher.java

```diff
@@ -95,6 +95,7 @@ public boolean isThreadSafe() {
   protected final DataFileVersionStrategy dstDataFileVersionStrategy;
   protected final boolean preserveDirModTime;
   protected final boolean resyncDirOwnerAndPermission;
+  protected final boolean requirePermissionSetForSuccess;
```

Comment:
nit: IMO sounds more complicated this way, when it seems pretty common sense that failure on setting perms should fail the job.  hence I'd personally reorient/reverse/rephrase to `shouldFailWhenPermissionFail` or `fatalPermsUpdateFailure`